base/terminology_processor.bal

```diff
@@ -14,206 +14,1476 @@
                         }
-                        
                     }
                 }
             }
+            return;
         }
-        return;
     }
 
     private isolated function conceptToCodeableConcept(
             CodeSystemConcept|ValueSetComposeIncludeConcept concept, uri system) returns CodeableConcept {
-        Coding codingValue = {
-            code: concept.code,
-            system: system
-        };
-        string? displayValue = concept.display;
-        if displayValue != () {
-            codingValue.display = displayValue;
-        }
-
-        CodeableConcept cConcept = {
-            coding: [
-                codingValue
-            ]
-        };
-
-        if concept is CodeSystemConcept {
-            string? defValue = concept.definition;
-            if defValue != () {
-                cConcept.text = defValue;
-            }
-        }
-        return cConcept;
-    } 
-
-    private isolated function conceptToCoding (CodeSystemConcept|ValueSetComposeIncludeConcept concept, uri system) returns Coding {
-
-        Coding codingValue = {
-            code: concept.code,
-            system: system
-        };
-        string? displayValue = concept.display;
-        if displayValue != () {
-            codingValue.display = displayValue;
-        }
-        return codingValue;
-    } 
+        lock {
+            Coding codingValue = {
+                code: concept.code,
+                system: system
+            };
+            string? displayValue = concept.display;
+            if displayValue != () {
+                codingValue.display = displayValue;
+            }
+
+            CodeableConcept cConcept = {
+                coding: [
+                    codingValue
+                ]
+            };
+
+            if concept is CodeSystemConcept {
+                string? defValue = concept.definition;
+                if defValue != () {
+                    cConcept.text = defValue;
+                }
+            }
+            return cConcept;
+        }
+
+    }
+
+    private isolated function conceptToCoding(CodeSystemConcept|ValueSetComposeIncludeConcept concept, uri system)
+                                                                                                            returns Coding {
+        lock {
+            Coding codingValue = {
+                code: concept.code,
+                system: system
+            };
+            string? displayValue = concept.display;
+            if displayValue != () {
+                codingValue.display = displayValue;
+            }
+            return codingValue;
+        }
+    }
+
+    private isolated function retrieveCodeSystemConcept(CodeSystem codeSystem, code|Coding concept) returns CodeSystemConcept? {
+        lock {
+            if concept is code {
+                CodeConceptDetails? conceptDetails = self.findConceptInCodeSystem(
+                codeSystem.clone(), concept.clone());
```

Comment:
Is there any specific reason to clone?

Reply:
Yes.. If we remove that getting the below error
```invalid attempt to transfer a value into a 'lock' statement with restricted variable usage```